### PR TITLE
Add user task support

### LINKED ISSUE #6116: Support User Tasks
**Is your feature request related to a problem? Please describe.**

As a user I would like to use a human task in my workflow. The user task should have a task form attached to it.

**Describe the solution you'd like**

XML:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<bpmn:definitions xmlns:bpmn="http://www.omg.org/spec/BPMN/20100524/MODEL" xmlns:zeebe="http://camunda.org/schema/zeebe/1.0" id="Definitions_1oquk3q">
  <bpmn:process id="user-task-process" name="User Task Process" isExecutable="true">
    <bpmn:extensionElements>
      <zeebe:userTaskForm id="userTaskForm_1">{"fields": [...]}</zeebe:userTaskForm>
    </bpmn:extensionElements>
    <bpmn:sequenceFlow id="Flow_1qwoyep" sourceRef="start" targetRef="user-task" />
    <bpmn:endEvent id="end" name="End">
      <bpmn:incoming>Flow_0xt42si</bpmn:incoming>
    </bpmn:endEvent>
    <bpmn:sequenceFlow id="Flow_0xt42si" sourceRef="user-task" targetRef="end" />
    <bpmn:userTask id="user-task" name="User Task">
      <bpmn:extensionElements>
        <zeebe:formDefinition formKey="camunda-forms:bpmn:userTaskForm_1" />
      </bpmn:extensionElements>
      <bpmn:incoming>Flow_1qwoyep</bpmn:incoming>
      <bpmn:outgoing>Flow_0xt42si</bpmn:outgoing>
    </bpmn:userTask>
    <bpmn:startEvent id="start" name="Start">
      <bpmn:outgoing>Flow_1qwoyep</bpmn:outgoing>
    </bpmn:startEvent>
  </bpmn:process>
</bpmn:definitions>
```

Implementation: 

The BPMN transformer transforms the `userTask` to a "pseudo" service task, with a given job type `humanTask` (TBD: Should we make this more transparent that his is a special job type?). The task form extension element is parsed into a task header for the job to be fetched by tasklist.

**Describe alternatives you've considered**

Implementing an own user task entity in Zeebe. I don't consider this right now as it's a bigger effort and the value of having it is unclear, until we decided that we want to support any specific API for user tasks in Zeebe itself. For now Tasklist API is our access point.

**Additional context**

The task form has to align with the Modeler team. The tasklist team has to know if we change the type and how the task header is called for the form.

### PR DESCRIPTION
**Pull request by menski**
## Description

Add support for BPMN User Task symbol in deployment. The user task has the same behavior as the service task internally. The job type will be set to `humanTask` and if an `formKey` is present it will be attached as task header with the name `io.camunda.zeebe::formKey`.

## Related issues

closes #6116 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [x] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [x] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Review by saig0 — APPROVED** [review]
Looks good :tada: 

I have only some minor suggestions/comments. Please have a look.

Besides that, the following things should be done eventually. Please create follow-up issues. 
* generate processes with user tasks
* add user tasks to the update tests 
* write documentation for user tasks
* mention user tasks in the release announcement

File: bpmn-model/src/test/java/io/zeebe/model/bpmn/validation/ZeebeUserTaskValidationTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    return new Object[][] {
+      {
+        Bpmn.createExecutableProcess("process")
+            .startEvent()
+            .userTask("task")
+            .zeebeFormKey("")
+            .endEvent()
+            .done(),
+        singletonList(expect(ZeebeFormDefinition.class, "Form key must be present and not empty"))
+      },
```
[the remark this answers is not in the document]
Comment (menski) [reply bpmn-model/src/test/java/io/zeebe/model/bpmn/validation/ZeebeUserTaskValidationTest.java:39]: `formKey` is optional, user can still use user tasks with no forms, i.e. just showing all variables

**Comment by menski** [comment]
Thanks Philipp for the fast review. I applied the review hints.

Follow-up issues:
- Docs: https://github.com/camunda-cloud/camunda-cloud-documentation/issues/199
- Update Test: https://github.com/camunda-cloud/zeebe/issues/6685
- Generate Processes with user task: https://github.com/camunda-cloud/zeebe/issues/6686

I also added a section for the user task to the release document, to be filled later.

File: protocol/src/main/java/io/zeebe/protocol/record/value/BpmnElementType.java [hunk 2]
```diff
@@ -75,6 +76,8 @@ public static BpmnElementType bpmnElementTypeFor(final String elementTypeName) {
         return BpmnElementType.SEQUENCE_FLOW;
       case "callActivity":
         return BpmnElementType.CALL_ACTIVITY;
+      case "userTask":
+        return BpmnElementType.USER_TASK;
```
[the remark this answers is not in the document]
Comment (menski) [reply protocol/src/main/java/io/zeebe/protocol/record/value/BpmnElementType.java:80]: I added a test for it

File: protocol/src/main/java/io/zeebe/protocol/Protocol.java [hunk 3]
```diff
@@ -62,6 +62,9 @@
    */
   public static final int KEY_BITS = 51;
 
+  public static final String USER_TASK_JOB_TYPE = "humanTask";
+  public static final String USER_TASK_FORM_KEY_HEADER_NAME = "io.camunda.zeebe:formKey";
```
[the remark this answers is not in the document]
Comment (menski) [reply protocol/src/main/java/io/zeebe/protocol/Protocol.java]: done https://github.com/camunda-cloud/zeebe/issues/6687

File: bpmn-model/src/main/java/io/zeebe/model/bpmn/builder/AbstractUserTaskBuilder.java [hunk 4]
```diff
@@ -40,4 +44,66 @@ public B implementation(final String implementation) {
...
+   * Creates an new user task form with the given context, assuming it is of the format
+   * camunda-forms and embedded inside the diagram.
+   *
+   * @param userTaskForm the XML encoded user task form json in the camunda-forms format
+   * @return the builder object
+   */
+  public B zeebeUserTaskForm(final String userTaskForm) {
+    final ZeebeUserTaskForm zeebeUserTaskForm = createZeebeUserTaskForm();
+    zeebeUserTaskForm.setTextContent(userTaskForm);
+    return zeebeFormKey("camunda-forms", "bpmn", zeebeUserTaskForm.getId());
```
[the remark this answers is not in the document]
Comment (menski) [reply bpmn-model/src/main/java/io/zeebe/model/bpmn/builder/AbstractUserTaskBuilder.java]: I put them into `ZeebeConstants` as we don't have access to the protocol in the bpmn model API

File: bpmn-model/src/main/java/io/zeebe/model/bpmn/builder/AbstractUserTaskBuilder.java [hunk 5]
```diff
@@ -40,4 +44,66 @@ public B implementation(final String implementation) {
...
+   *
+   * @param userTaskFormKey the key of the user task form element
+   * @param userTaskForm the XML encoded user task form json in the camunda-forms format
+   * @return the builder object
+   */
+  public B zeebeUserTaskForm(final String userTaskFormKey, final String userTaskForm) {
+    final ZeebeUserTaskForm zeebeUserTaskForm = createZeebeUserTaskForm();
+    zeebeUserTaskForm.setId(userTaskFormKey);
+    zeebeUserTaskForm.setTextContent(userTaskForm);
+    return zeebeFormKey("camunda-forms", "bpmn", userTaskFormKey);
```
[the remark this answers is not in the document]
Comment (menski) [reply bpmn-model/src/main/java/io/zeebe/model/bpmn/builder/AbstractUserTaskBuilder.java]: I choose `id` to keep it simple and consistent

**Pull request merged**